types/index.d.ts

```diff
@@ -542,35 +542,17 @@ declare class Realm {
      * @param  {Realm.UpdateMode} mode? If not provided, `Realm.UpdateMode.Never` is used.
      * @returns T & Realm.Object
      */
-    create<T>(type: string, properties: RealmInsertionModel<T>, mode?: Realm.UpdateMode): T & Realm.Object
+    create<T>(type: string, properties: RealmInsertionModel<T>, mode?: Realm.UpdateMode.Never): T & Realm.Object;
```

Comment:
Why not remove the overloads and use this instead 
create<T>(type: string, properties: RealmInsertionModel<T>, mode: Realm.UpdateMode = Realm.UpdateMode.Never): T & Realm.Object;

In TS default-initialized parameters at the end are treated as optional

Reply:
I had the same thought, and it would simplify the declaration - but default initializers are only allowed in implementations, not (pure) declarations 🤷‍♂️ - I'm not entirely sure why this is a limit... as you'll need to know the underlying implementation anyway, when writing the declarations..?

TS error says:
```
A parameter initializer is only allowed in a function or constructor implementation.
```

Reply:
🤦 got it.  
But then this is a good reminder that  we don't need the initializer since this is just a type check. We could just write this 
`create<T>(type: string, properties: Partial<T> | Partial<RealmInsertionModel<T>> | RealmInsertionModel<T>, mode?: Realm.UpdateMode): T & Realm.Object;`

Reply:
Yeah, but this would allow inserts of `Partial<T>` for `Realm.UpdateMode.Never`. This is where we want TS to help the user uphold their own model definitions (basically mandatory/optional props) - that's why I kept the two separate.

Reply:
I see.